rpcs3/main.cpp

```diff
@@ -494,27 +522,39 @@ int main(int argc, char** argv)
 	{
 		sys_log.notice("Booting application from command line: %s", args.at(0).toStdString());
 
+		auto game = sstr(args.at(0));
+
 		// Propagate command line arguments
 		std::vector<std::string> argv;
 
-		if (args.length() > 1)
+		// we skip the first entry (which should be the game)
+		for (int i = 1; i < args.length(); i++)
 		{
-			argv.emplace_back();
-
-			for (int i = 1; i < args.length(); i++)
-			{
-				const std::string arg = args[i].toStdString();
-				argv.emplace_back(arg);
-				sys_log.notice("Optional command line argument %d: %s", i, arg);
-			}
+			const std::string arg = args[i].toStdString();
+			argv.emplace_back(arg);
+			sys_log.notice("Optional command line argument %d: %s", i, arg);
 		}
 
 		// Ugly workaround
-		QTimer::singleShot(2, [config_override_path, path = sstr(QFileInfo(args.at(0)).absoluteFilePath()), argv = std::move(argv)]() mutable
+		QTimer::singleShot(2, [&app, config_override_path, game, argv = std::move(argv), exit_after]() mutable
 		{
 			Emu.argv = std::move(argv);
 			Emu.SetForceBoot(true);
-			Emu.BootGame(path, "", true);
+			if (game.starts_with('['))
+			{
+				// we'll boot from title
+				game = game.substr(1, game.find("]")-1);
+				Emu.BootGameFromTitleId(game, true);
+			}
+			else
+			{
+				// we'll boot from path
+				game = sstr(QFileInfo(QString::fromStdString(game)).absoluteFilePath());
+				Emu.BootGameFromPath(game, "", true);
+			}
+
+			if (exit_after)
+				app->exit(0);
```

Comment:
This is wrong in so many ways....
A: It's just wrong
B: We have our own exit function.
C: Emu,BootGame is not blocking, so this makes no sense whatsoever
D: The whole exit_after arg cannot be handled from outside, it has to be handled by the Emulator